builtin/receive-pack.c

```diff
@@ -1118,7 +1130,7 @@ static const char *update(struct command *cmd, struct shallow_info *si)
 	}
 
 	if (do_update_worktree) {
-		ret = update_worktree(new_oid->hash);
+		ret = update_worktree(new_oid->hash, find_shared_symref("HEAD", name));
```

Comment:
did I assigned `const struct worktree *worktree` correctly in this statement ?? @dscho

Reply:
Yes, you did. The problem would appear to be that `find_shared_symref()` might mis-identify the current directory for the main worktree. That would be worth investigating. The question is whether you feel up for it...

Reply:
I'm up for it.

Reply:
Can you follow the code and see where it gets its (absolute) path from? That should give you an idea where it goes wrong: the worktree should be `error: work_tree: /home/harry/Desktop/git/t/trash directory.t5516-fetch-push/testrepo` (note the extra `/.git` in the values you reported).